Sources/AppStoreServerLibrary/AppStoreServerAPIClient.swift

```diff
@@ -7,54 +7,47 @@ import AsyncHTTPClient
 import NIOHTTP1
 import NIOFoundationCompat
 
-public class AppStoreServerAPIClient {
+// MARK: - HTTP Client Protocol
+
+/// Protocol for HTTP clients that work with App Store Server API
+public protocol AppStoreHTTPClient: Sendable {
+    /// Execute an HTTP request
+    func execute(
+        _ request: HTTPClientRequest,
+        timeout: Duration
+    ) async throws -> HTTPClientResponse
+}
 
-    public enum ConfigurationError: Error {
-        /// Xcode is not a supported environment for an AppStoreServerAPIClient
-        case invalidEnvironment
+extension HTTPClient: AppStoreHTTPClient {
+    /// Execute HTTP request
+    /// - Parameters:
+    ///   - request: HTTP request
+    ///   - timeout: If execution is idle for longer than timeout then throw error
+    /// - Returns: HTTP response
+    public func execute(
+        _ request: HTTPClientRequest,
+        timeout: Duration
+    ) async throws -> HTTPClientResponse {
+        return try await self.execute(request, timeout: .init(timeout))
     }
-    
-    private static let userAgent = "app-store-server-library/swift/3.1.0"
-    private static let productionUrl = "https://api.storekit.itunes.apple.com"
-    private static let sandboxUrl = "https://api.storekit-sandbox.itunes.apple.com"
-    private static let localTestingUrl = "https://local-testing-base-url"
-    private static let appStoreConnectAudience = "appstoreconnect-v1"
-    
-    private let signingKey: P256.Signing.PrivateKey
-    private let keyId: String
-    private let issuerId: String
-    private let bundleId: String
-    private let url: String
-    private let client: HTTPClient
+}
+
+public final class AppStoreServerAPIClient: Sendable {
+
+    public static let userAgent = "app-store-server-library/swift/3.1.0"
+    public static let appStoreConnectAudience = "appstoreconnect-v1"
+    private let config: AppStoreServerAPIConfiguration
+    private let client: AppStoreHTTPClient
     ///Create an App Store Server API client
     ///
-    ///- Parameter signingKey: Your private key downloaded from App Store Connect
-    ///- Parameter issuerId: Your issuer ID from the Keys page in App Store Connect
-    ///- Parameter bundleId: Your app’s bundle ID
-    ///- Parameter environment: The environment to target
-    public init(signingKey: String, keyId: String, issuerId: String, bundleId: String, environment: AppStoreEnvironment) throws {
```

Comment:
This public API has to stay around, as removing it is a breaking change. It could call into the other initializer though.